src/App/screens/Instructor/components/LessonList/index.js

```diff
@@ -0,0 +1,34 @@
+import React from 'react'
+import {connect} from 'react-redux'
+import LessonsByPage from '../LessonsByPage'
+import PaginatedLessonList from './components/PaginatedLessonList'
+
+const LessonList = ({
+  instructor,
+  lessonPage,
+  states,
+  isOwnedByInstructor = false,
+}) => (
+  <LessonsByPage
+    instructor={isOwnedByInstructor ? instructor : false}
+    states={states}
+  >
+    {({currentPage, fetchLessons, pageSize}) => (
+      <PaginatedLessonList
+        lessons={lessonPage.lessons}
+        fetchLessons={fetchLessons}
+        total={lessonPage.total}
+        currentPage={currentPage}
+        pageSize={pageSize}
+      />
+    )}
+  </LessonsByPage>
+)
+
+export default connect(
+  ({instructorScreen}) => ({
+    instructor: instructorScreen.instructor,
+    lessonPage: instructorScreen.lessonPage,
+  }),
+  null,
+)(LessonList)
```

Comment:
I'm proud of this thing. Reduced four other files into this which owns all the lesson lists now.